src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -4284,18 +4197,24 @@ private String layoutChars(boolean sci) {
             // currency fast path
             int lowInt = (int)intCompact % 100;
             int highInt = (int)intCompact / 100;
-            return (Integer.toString(highInt) + '.' +
-                    StringBuilderHelper.DIGIT_TENS[lowInt] +
-                    StringBuilderHelper.DIGIT_ONES[lowInt]) ;
+            int highIntSize = DecimalDigits.stringSize(highInt);
+            byte[] buf = new byte[highIntSize + 3];
+            DecimalDigits.putPairLatin1(buf, highIntSize + 1, lowInt);
+            buf[highIntSize] = '.';
+            DecimalDigits.getCharsLatin1(highInt, highIntSize, buf);
+            try {
+                return JLA.newStringNoRepl(buf, StandardCharsets.ISO_8859_1);
+            } catch (CharacterCodingException cce) {
+                throw new AssertionError(cce);
+            }
         }
 
-        StringBuilderHelper sbHelper = new StringBuilderHelper();
         char[] coeff;
         int offset;  // offset is the starting index for coeff array
         // Get the significand as an absolute value
         if (intCompact != INFLATED) {
-            offset = sbHelper.putIntCompact(Math.abs(intCompact));
-            coeff  = sbHelper.getCompactCharArray();
+            coeff = new char[19];
```

Comment:
A possibility here would be to change `coeff` to be a String. The “else” branch already creates a string and has to additionally create a char array from it. If this is the only place where the `DecimalDigits.getChars(… char[])`, some extra code duplication could be removed. The `offset` could then also be removed